zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/UsageMetricHandler.java

```diff
@@ -48,57 +59,145 @@ public boolean handlesRecord(final Record<UsageMetricRecordValue> record) {
   @Override
   public List<String> generateIds(final Record<UsageMetricRecordValue> record) {
     final long key = record.getKey();
-    return record.getValue().getCounterValues().keySet().stream()
+    final UsageMetricRecordValue value = record.getValue();
+
+    return Stream.concat(
+            value.getCounterValues().keySet().stream(), value.getSetValues().keySet().stream())
         .map(tenantId -> ID_PATTERN.formatted(key, tenantId))
         .toList();
```

Comment:
💭 I'm not sure if it's correct to return all the IDs if we use a batch object.

By looking at the code it looks like the [`updateAndCacheEntity`](https://github.com/camunda/camunda/blob/main/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/ExporterBatchWriter.java#L56-L57) will be called for each ID.

For each ID it calls `createNewEntity` so we create a new `UsageMetricsBatch`. Then it calls the `updateEntity` where in every one of them we add all the usage metric entities but the same ones multiple times.

I feel like this should only happen once, so we need to return one ID, probably the `key` only.